components/script/dom/documentfragment.rs

```diff
@@ -60,6 +60,38 @@ impl DocumentFragment {
     pub fn id_map(&self) -> &DomRefCell<HashMapTracedValues<Atom, Vec<Dom<Element>>>> {
         &self.id_map
     }
+
+    /// <https://html.spec.whatwg.org/multipage/#fragment-serializing-algorithm-steps>
+    pub fn fragment_serialization_algorithm(&self, require_well_formed: bool) -> DOMString {
+        // Step 1. Let context document be node's node document.
+        let context_document = document_from_node(self);
+
+        // Step 2. If context document is an HTML document, return the result of HTML fragment serialization algorithm
+        // with node, false, and « ».
+        let mut writer = vec![];
+        if context_document.is_html_document() {
+            html5ever::serialize(
```

Comment:
Can we make https://github.com/servo/servo/blob/main/components/script/dom/element.rs#L1330 a helper we can reuse here instead?